### PR TITLE
Add list case note gateway method

### PR DESCRIPTION
**Pull request by Renny4Real**
### *What is the problem we're trying to solve*

Creating an endpoint to list historic case note data and make it available via the interim social care solution.
[JIRA TICKET](https://hackney.atlassian.net/browse/SCS-540)

### *What changes have we introduced*

This creates a gateway method that when called will query the database and
return a list of objects with some of their fields populated.

Only some fields are populated per object as the UI where this LIST endpoint will be called
displays summary fields of each record a person has.

#### _Checklist_

- [X] Added tests to cover all new production code
- [X] Checked all code for possible refactoring

### *Follow up actions after merging PR*

- Focusing on getting the endpoint working end-to-end to test the connection with the [Service API](https://github.com/LBHackney-IT/social-care-case-viewer-api)
- Will work on making any changes to what the gateway returns after testing the connection with the [Service API](https://github.com/LBHackney-IT/social-care-case-viewer-api)

### CODE REVIEW

File: ResidentsSocialCarePlatformApi.Tests/V1/Gateways/SocialCare/GetCaseNotesTests.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        }
+
+        [Test]
+        public void WhenThereIsOneMatch_ReturnsCaseNoteInformationForAGivenPersonId()
+        {
+            var person = AddPersonToDatabase();
+            AddCaseNoteToDatabase(person.Id);
+
+            var response = _classUnderTest.GetCaseNotes(person.Id);
+            response.FirstOrDefault().Should().NotBe(null);
```
Comment (chubberlisk) [inline ResidentsSocialCarePlatformApi.Tests/V1/Gateways/SocialCare/GetCaseNotesTests.cs]: For readability, could we add a newline to separate the "act" and "assertion" of this test?

```suggestion
            var response = _classUnderTest.GetCaseNotes(person.Id);
            
            response.FirstOrDefault().Should().NotBe(null);
```

File: ResidentsSocialCarePlatformApi.Tests/V1/Infrastructure/SocialCareContextTests.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+namespace ResidentsSocialCarePlatformApi.Tests.V1.Infrastructure
+{
+    [TestFixture]
+    public class SocialCareContextTests : DatabaseTests
+    {
+        [Test]
+        public void CanCreateADatabaseRecordForAPerson()
+        {
+            var databaseEntity = TestHelper.CreateDatabasePersonEntity();
```
Comment (chubberlisk) [inline ResidentsSocialCarePlatformApi.Tests/V1/Infrastructure/SocialCareContextTests.cs]:
```suggestion
            var person = TestHelper.CreateDatabasePersonEntity();
```

File: ResidentsSocialCarePlatformApi/V1/Factories/EntityFactory.cs [hunk 3]
```diff
@@ -49,6 +49,22 @@ public static Domain.Address ToDomain(this Infrastructure.Address address)
             };
         }
 
+        public static List<CaseNoteInformation> ToDomain(this IEnumerable<CaseNote> caseNotes)
```
Comment (chubberlisk) [inline ResidentsSocialCarePlatformApi/V1/Factories/EntityFactory.cs:52]: Nice - having separate ones for listing and viewing!

**Review by chubberlisk — APPROVED** [review]
Looks good to me - just two very smol readability comments! 🚢

File: ResidentsSocialCarePlatformApi.Tests/V1/Gateways/SocialCare/GetCaseNotesTests.cs (hunk 1 above)
Comment (Renny4Real) [reply ResidentsSocialCarePlatformApi.Tests/V1/Gateways/SocialCare/GetCaseNotesTests.cs]: 👍🏾

File: ResidentsSocialCarePlatformApi.Tests/V1/Infrastructure/SocialCareContextTests.cs (hunk 2 above)
Comment (Renny4Real) [reply ResidentsSocialCarePlatformApi.Tests/V1/Infrastructure/SocialCareContextTests.cs]: 👍🏾

**Pull request merged**